### PR TITLE
add rel='nofollow' to user links

### PR DESCRIPTION
**Pull request by escattone**
mozilla/sumo#1153

This PR adds `rel="nofollow"` to all user links.

### CODE REVIEW

File: kitsune/sumo/jinja2/layout/breadcrumbs.html [hunk 1]
```diff
@@ -3,7 +3,12 @@
   <ol id="breadcrumbs" class="breadcrumbs--list">
   {% for target, label in breadcrumbs %}
     {% if target %}
-      <li><a href="{{ target }}">{{ label|truncate(length=30) }}</a></li>
+      {% if target is string %}
```
Comment (akatsoulas) [inline kitsune/sumo/jinja2/layout/breadcrumbs.html:6]: Why are you introducing this change here? Even if it is a valid change, shall we include it in a new PR since it does not seem related to the the `nofollow` directive.

**Review by akatsoulas — COMMENTED** [review]
Overall this is looking good but I do have a couple of comments:

* This PR seems very similar to #5320. They do share some common functionality and more or less there are about the same thing. I believe we should tackle everything in a single PR and move the work from here to #5320 and clean that PR up
* Why did you remove `ugc`?
* This PR applies the `nofollow` directive to user related links. Though there is some overlap with the use of the same directive in base pages. For example in the kbforums in PR 5320 you add the `nofollow` which is also applied in the user links of `discussions.html`. I do understand that you are focusing here on the user links but I am also wondering if there is a way to have some guidelines/rules on how we apply this. Eg first on the base.html of each app if applicable, second to specific pages if we do not want it on the base.html etc. Some kind of policy
* We need to document that policy in the documentation

File: kitsune/sumo/jinja2/layout/breadcrumbs.html (hunk 1 above)
Comment (escattone) [reply kitsune/sumo/jinja2/layout/breadcrumbs.html:6]: This is needed to support the new breadcrumbs format, which allows extra attributes (in this case `rel="nofollow"`) to be attached to breadcrumbs specified by sub-classes, for example, `kitsune/users/jinja2/users/edit_profile.html` below.
